src/pages/Approvals/ApprovalItemsPicker.tsx

```diff
@@ -0,0 +1,171 @@
+import { useFieldArray, Controller, useFormContext } from "react-hook-form";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faClose } from "@fortawesome/free-solid-svg-icons";
+import styled from "styled-components";
+
+import { makeLabeledInput } from "components/Inputs/LabeledInput";
+import { ApprovalItemType, PartialApprovalItem } from "apiClient/approvals";
+import { Select } from "components/Select";
+import { GearTypeSelect } from "components/GearTypeSelect";
+import { GearItemSelect } from "components/GearItemSelect";
+
+import { FormValues } from "./types";
+
+const LabeledInput = makeLabeledInput<FormValues>();
+
+export function ApprovalItemsPicker() {
+  const formObject = useFormContext<FormValues>();
+  const {
+    fields: itemFields,
+    append,
+    remove,
+  } = useFieldArray({
+    control: formObject.control,
+    name: "items",
+  });
+  const items = formObject.watch("items");
+  return (
+    <fieldset>
+      Items approved:
+      {itemFields.map((field, index) => {
+        return (
+          <StyledItem key={field.id} className="p-2 mb-3 mt-3">
+            <div className="d-flex justify-content-between mb-3">
+              <strong>Item #{index + 1}</strong>
+              {items.length > 1 && (
+                <button
+                  type="button"
+                  className="btn btn-outline-secondary"
+                  onClick={() => remove(index)}
+                  style={{
+                    borderColor: "#ced4da",
+                  }}
+                >
+                  <FontAwesomeIcon icon={faClose} />
+                </button>
+              )}
+            </div>
+            <Controller
+              control={formObject.control}
+              name={`items.${index}.type`}
+              rules={{ required: true }}
+              render={({ field: { onChange, onBlur, value, ref } }) => {
+                return (
+                  <ApprovalTypePicker
+                    value={value}
+                    onChange={(v) => {
+                      formObject.resetField(`items.${index}`);
+                      onChange(v);
+                    }}
+                    onBlur={onBlur}
+                  />
+                );
+              }}
+            />
+            {items[index].type === ApprovalItemType.gearType ? (
+              <>
+                <LabeledInput
+                  title="Type:"
+                  name={`items.${index}.item.gearType` as const}
+                  renderComponent={({ value, onChange, invalid }) => {
+                    return (
+                      <GearTypeSelect
+                        value={value ?? null}
+                        onChange={(val) => onChange(val?.id)}
+                        invalid={invalid}
+                      />
+                    );
+                  }}
+                  options={{
+                    required: true,
+                  }}
+                />
+                <LabeledInput
```

Comment:
Does this automatically default to 1? Might make sense to add a default value because this isn't required.

Reply:
Yes, the default values are defined further down in the file in `defaultItem`